cabal-ghc921.project

```diff
@@ -36,11 +54,38 @@ package *
 
 write-ghc-environment-files: never
 
-index-state: 2021-12-29T12:30:08Z
+index-state: 2022-01-03T18:45:00Z
 
 constraints:
   -- These plugins doesn't work on GHC92 yet
-  haskell-language-server +ignore-plugins-ghc-bounds -brittany -class -fourmolu -splice -stylishhaskell -tactic -refineImports -callhierarchy -retrie
+  haskell-language-server
+    +ignore-plugins-ghc-bounds
+    -alternateNumberFormat
+    -brittany
+    -callhierarchy
+    -class
+    -eval
+    -floskell
+    -fourmolu
+    -haddockComments
+    -hlint
+    -moduleName
+    -ormolu
+    -qualifyImportedNames
+    -refineImports
+    -retrie
+    -splice
+    -stylishhaskell
+    -tactic,
+  ghc-lib-parser ^>= 9.2,
+  attoparsec ^>= 0.14.3,
+  ghc-exactprint >= 1.3,
+  retrie >= 1.2,
+  direct-sqlite == 2.3.26,
+  lens >= 5.0.1,
+  primitive-unlifted ==0.1.3.1,
+  -- these constraints are for head.hackage
+  aeson ==1.5.6.0,
```

Comment:
On <sub>my</sub> machine:
```
Linux 5.15.12 x86_64 GNU/Linux
```
`aeson ==1.5.6.0 && lens >= 5.0.1` continuously:
```bash
cabal v2-build --project-file cabal-ghc921.project
Starting     lens-5.0.1 (lib)
Starting     aeson-1.5.6.0 (lib)
Building     lens-5.0.1 (lib)
Building     aeson-1.5.6.0 (lib)

...

[40 of 86] Compiling Control.Lens.Wrapped ( src/Control/Lens/Wrapped.hs, dist/build/Control/Lens/Wrapped.o, dist/build/Control/Lens/Wrapped.dyn_o )

src/Control/Lens/Wrapped.hs:756:15: error:
    Not in scope: type constructor or class ‘S.Option’
    Module ‘Data.Semigroup’ does not export ‘Option’.
    |
756 | instance (t ~ S.Option b) => Rewrapped (S.Option a) t
    |               ^^^^^^^^

...

src/Control/Lens/Wrapped.hs:759:31: error:
    Not in scope: data constructor ‘S.Option’
    Module ‘Data.Semigroup’ does not export ‘Option’.
    |
759 |   _Wrapped' = iso S.getOption S.Option
    |                               ^^^^^^^^
cabal: Failed to build aeson-1.5.6.0 (which is required by
hls-test-utils-1.1.0.2, hls-pragmas-plugin-1.0.1.1 and others). See the build
log above for details.
Failed to build lens-5.0.1 (which is required by hls-test-utils-1.1.0.2,
hls-pragmas-plugin-1.0.1.1 and others). See the build log above for details.
```

Also note that `head.hackage` recently got [patches](https://gitlab.haskell.org/ghc/head.hackage/-/commit/84cad4c5c7fe1f0a6b3e15a02cc08af9ce26fe31) to `aeson` `1.5.6` and `2.0.2`.

`--allow-newer` for all packages worked, while `allow-newer&older` `aeson&lens` results in the same versions picked, so `lens` version choice also gets restricted elsewhere.

Reply:
`aeson-1.5.6.0` is not compatible with GHC 9.2, unless you apply head.hackage patches. It would help if someone makes HLS compatible with `aeson-2.0.3.0`, this could be done independently of this PR.

> Also note that `head.hackage` recently got [patches](https://gitlab.haskell.org/ghc/head.hackage/-/commit/84cad4c5c7fe1f0a6b3e15a02cc08af9ce26fe31) to `aeson` `1.5.6` and `2.0.2`.

These are irrelevant to GHC 9.2, they are for GHC HEAD + bytestring-0.11.2.0.

Reply:
>  It would help if someone makes HLS compatible with aeson-2.0.3.0, this could be done independently of this PR.

We are working on that here: #2551

Reply:
<sub>Yes (that particular patch maybe not relevant), but there are several recent patches, so updated to more precise name&link</sub>

The central & the first question is, why current HLS package configuration does not build, at least for some people who try to build it locally.

Reply:
To recap: 
the central & the first question is, why the current HLS package configuration in this PR does not build, at least for some people who try to build it locally.

Reply:
CI status, build seems to work...

Would be looked deeper as would be considered to be reopened as a report after the merge of the code.

Reply:
Seems (currently building) the solution was that PR needs:

```
cabal v2-update --project-file=cabal-ghc921.project
```

Opened https://github.com/haskell/haskell-language-server/issues/2574, because CI needs to do updates this way.